### ISSUE TITLE
Can not find module error (Nightly 600 - Safari 12)

### ISSUE THREAD

**Issue by ittaibaratz** [parcel-bundler/parcel#5888]
# 🐛 bug report

After upgrading Parcel to the latest nightly, we are having problems with Safari 12 (Via SauceLabs).
Everything runs fine in other browsers, including Safari 14.

![image](https://user-images.githubusercontent.com/1454683/108526509-12b57c80-7286-11eb-8e2a-d2ef9c906d7c.png)

## 🎛 Configuration (.babelrc, package.json, cli command)

Babel:
```
module.exports = function (api) {
  api.cache(true);

  return {
    presets: [
      [
        '@babel/preset-env',
        {
          bugfixes: true,
          corejs: 2,
          useBuiltIns: 'entry'
        }
      ],
      '@babel/preset-react',
      '@babel/preset-typescript'
    ],
    plugins: [
      '@babel/plugin-proposal-class-properties',
      ["@babel/plugin-proposal-decorators", {decoratorsBeforeExport: true}]
    ]
  }
};
```

Our entry is index.html which has a few JS resources:
```
    <script src="./polyfills.js"></script>
    <script src="./react.js"></script>
    <!-- Placeholder for insertion of shell js bundle -->
    <script src="./main.js"></script>
    <script defer src="./browser-support.js"></script>
    <script defer src="./error-page.js"></script>
```

polyfills includes core-js polyfills as well as some initilization code and runs fine.
react includes React/ReactDOM and runs fine.
main.js is where our App component lives, and is where the problem above happens.
 
## 🤔 Expected Behavior

Bundled app should load in Safari 12.

## 😯 Current Behavior

Bundled app does not load correctly.

Note that when running a dev server, it does run fine - The problem is only with the build.

## 💻 Code Sample

I can provide access to @devongovett (Same company). Can also try to do a minimal reproduction if required.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0 nightly 577 |
| Node             | 14/15 (latest) |
| npm/Yarn         | latest | 
| Operating System | CentOS / Mac OS |

**Comment by ross-pfahler** [comment]
Here's a small repro setup: https://github.com/ross-pfahler/parcel-2-demos/tree/multi-file-html-entry. it is not just safari.

I think the key is we are using multiple files in an entry html file. So, it looks like this:
```
...
  <body>
    <div id="app"></div>
    <script src="./polyfills.ts"></script>
    <script src="./react.ts"></script>
    <script src="./index.tsx"></script>
  </body>
...
```

because of parcel's ability to generate long term hashes for files, we get really good cache hit-rates on files that don't change a lot; in this case, these are two libraries containing dependencies we need to load our app (in the example i chose react + core-js, but could be anything).

Our app entry (index.tsx) is deployed daily, but we dont update our external dependencies as often. This means our users get good cache hits on polyfill.ts and react.ts, and keeps index.tsx smaller since it varies more often.

Lastly, with http/2 browsers generally download all of these files in parallel (although execution in script insertion order). Keeping them small ensures we arent stuck waiting for one big file that changes often.

**Comment by mischnic** [comment]
Your specific reproduction is actually the same issue as https://github.com/parcel-bundler/parcel/issues/4858. But the same bug might also cause the original error

**Comment by ittaibaratz** [comment]
Discussed this with @devongovett and the issue is because we have a dynamically loaded script that depends (via bundling) on a deferred script (using defer) which may or may not load in time. 

We'll solve on our end, this issue can be closed now.
